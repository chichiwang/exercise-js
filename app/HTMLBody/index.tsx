import React from 'react';

import Head from 'app/Head';

interface HTMLBodyProps extends React.PropsWithChildren {}

function HTMLBody({ children }: HTMLBodyProps) {
  return (
    <html lang="en">
      <Head />
      <body>
        { children }
      </body>
    </html>
  );
}

export default HTMLBody;
