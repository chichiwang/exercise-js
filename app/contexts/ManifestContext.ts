import { createContext } from 'react';

import { Manifest } from 'app/types';

const ManifestContext = createContext<Manifest>({});

export default ManifestContext;
