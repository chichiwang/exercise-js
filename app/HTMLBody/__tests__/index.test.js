import React from 'react';
import renderer from 'react-test-renderer';

import ManifestContext from 'app/contexts/ManifestContext';

import HTMLBody from '../index';

function ChildComponent() {
  return (
    <h1>Body Contents</h1>
  );
}

function MockedHead() {
  return (
    <head>
      <title>HTMLBody Test</title>
    </head>
  );
}

jest.mock('app/Head', () => MockedHead);

const mockedManifest = {
  'app.css': 'HTML Body CSS path',
};

describe('HTMLBody Component', () => {
  test('matches no-child snapshot', () => {
    const tree = renderer.create(
      <ManifestContext.Provider value={mockedManifest}>
        <HTMLBody />
      </ManifestContext.Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches child snapshot', () => {
    const tree = renderer.create(
      <ManifestContext.Provider value={mockedManifest}>
        <HTMLBody>
          <ChildComponent />
        </HTMLBody>
      </ManifestContext.Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
