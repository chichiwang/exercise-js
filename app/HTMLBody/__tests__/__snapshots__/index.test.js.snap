// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTMLBody Component matches child snapshot 1`] = `
<html
  lang="en"
>
  <head>
    <title>
      HTMLBody Test
    </title>
  </head>
  <body>
    <h1>
      Body Contents
    </h1>
  </body>
</html>
`;

exports[`HTMLBody Component matches no-child snapshot 1`] = `
<html
  lang="en"
>
  <head>
    <title>
      HTMLBody Test
    </title>
  </head>
  <body />
</html>
`;
